### PR TITLE
Add formatting checks.

### PR DESCRIPTION
**Pull request by nateglims**

### CODE REVIEW

**Review by sarenameas — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -72,3 +72,21 @@ jobs:
         run: |
           PATH=$PATH:$PWD/tools/spell
           for lexfile in `find . -name lexicon.txt`; do dir=${lexfile%/lexicon.txt}; find-unknown-comment-words --directory $dir; if [ $? -ne "0" ]; then exit 1; fi; done;
+  fmt:
```
Comment (aggarw13) [inline .github/workflows/ci.yml:75]: Wouldn't "formatting-check" be more descriptive?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -72,3 +72,21 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Uncrustify
+        run: sudo apt-get install uncrustify
+      - name: Run Uncrustify
+        run: find . -iname "*.c" -exec uncrustify --check -c tools/uncrustify.cfg {} +
```
Comment (aggarw13) [inline .github/workflows/ci.yml:82]: This seems to be limited the check only on source files. What about the header files?
```suggestion
        run: find . -iname "*.[hc]" -exec uncrustify --check -c tools/uncrustify.cfg {} +
```

**Review by aggarw13 — CHANGES_REQUESTED** [review]
Please update `find` command to also look for header files

**Comment by sarenameas** [comment]
> Please update `find` command to also look for header files

It is finding header files: `"*.[hc]"`

**Review by sarenameas — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nateglims) [reply .github/workflows/ci.yml]: Changed to formatting. Check will be redundant with the workflow name.

**Review by aggarw13 — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aggarw13) [reply .github/workflows/ci.yml]: Was recommending in reference to the `spell-check` job name, though I agree with your point of redundancy

**Review by sarenameas — APPROVED** [review]

**Pull request merged**